src/main/java/org/mifos/connector/bulk/camel/routes/FileProcessingRoute.java

```diff
@@ -0,0 +1,110 @@
+    private static final String OVERRIDE_HEADER = "overrideHeader";
+
+    @Autowired
+    private CsvMapper csvMapper;
+
+    @Override
+    public void configure() throws Exception {
+
+        /**
+         * Parse the [Transaction] array from the csv file
+         * exchangeInput: [LOCAL_FILE_PATH] the absolute path to the csv file
+         * exchangeOutput: [TRANSACTION_LIST] containing the list of [Transaction]
+         */
+        from("direct:get-transaction-array")
+                .id("direct:get-transaction-array")
+                .log("Starting route direct:get-transaction-array")
+                .process(exchange -> {
+                    Double totalAmount = 0.0;
+                    Long failedAmount = 0L;
+                    Long completedAmount = 0L;
+                    String filename = exchange.getProperty(FILE_NAME, String.class);
+                    CsvSchema schema = CsvSchema.emptySchema().withHeader();
+                    FileReader reader = new FileReader(filename);
+                    MappingIterator<Transaction> readValues = csvMapper.readerWithSchemaFor(Transaction.class).with(schema).readValues(reader);
+                    List<Transaction> transactionList = new ArrayList<>();
+                    while (readValues.hasNext()) {
+                        Transaction current = readValues.next();
+                        transactionList.add(current);
+                        totalAmount += Double.parseDouble(current.getAmount());
+                    }
+                    reader.close();
+                    exchange.setProperty(TRANSACTION_LIST, transactionList);
+                    exchange.setProperty(TRANSACTION_LIST_LENGTH, transactionList.size());
+                    exchange.setProperty(TOTAL_AMOUNT, totalAmount);
+                    exchange.setProperty(ONGOING_AMOUNT, totalAmount); // initially ongoing amount is same as total amount
+                    exchange.setProperty(FAILED_AMOUNT, failedAmount);
+                    exchange.setProperty(COMPLETED_AMOUNT, completedAmount);
+                });
+
+        /**
+         * updates the data in local file
+         * exchangeInput:
+         *      [LOCAL_FILE_PATH] the absolute path to the csv file
+         *      [RESULT_TRANSACTION_LIST] containing the list of [Transaction]
+         *      [OVERRIDE_HEADER] if set to true will override the header or else use the existing once in csv file
+         */
+        from("direct:update-result-file")
+                .id("direct:update-result-file")
+                .log("Starting route direct:update-result-file")
+                .process(exchange -> {
+                    String filepath = exchange.getProperty(LOCAL_FILE_PATH, String.class);
+                    List<TransactionResult> transactionList = exchange.getProperty(RESULT_TRANSACTION_LIST, List.class);
+
+                    for(TransactionResult result : transactionList){
+                        logger.info(result.toString());
```

Comment:
This looks like a debug step. remove this

Reply:
I am typically removing these log statements or changing these log statements to debug level once all PR review comments are addressed.